planning/behavior_path_goal_planner_module/src/util.cpp

```diff
@@ -78,13 +79,62 @@ lanelet::ConstLanelets getPullOverLanes(
 
 lanelet::ConstLanelets generateExpandedPullOverLanes(
   const RouteHandler & route_handler, const bool left_side, const double backward_distance,
-  const double forward_distance, double bound_offset)
+  const double forward_distance, const double bound_offset)
 {
   const auto pull_over_lanes =
     getPullOverLanes(route_handler, left_side, backward_distance, forward_distance);
 
   return left_side ? lanelet::utils::getExpandedLanelets(pull_over_lanes, bound_offset, 0.0)
-                   : lanelet::utils::getExpandedLanelets(pull_over_lanes, 0.0, bound_offset);
+                   : lanelet::utils::getExpandedLanelets(pull_over_lanes, 0.0, -bound_offset);
+}
+
+static double getOffsetToLanesBoundary(
+  const lanelet::ConstLanelets & lanelet_sequence, const geometry_msgs::msg::Pose target_pose,
+  const bool left_side)
+{
+  lanelet::ConstLanelet closest_lanelet;
+  lanelet::utils::query::getClosestLanelet(lanelet_sequence, target_pose, &closest_lanelet);
+
+  // the boundary closer to ego. if left_side, take right boundary
+  const auto & boundary3d = left_side ? closest_lanelet.rightBound() : closest_lanelet.leftBound();
+  const auto boundary = lanelet::utils::to2D(boundary3d);
+  using lanelet::utils::conversion::toLaneletPoint;
+  const auto arc_coords = lanelet::geometry::toArcCoordinates(
+    boundary, lanelet::utils::to2D(toLaneletPoint(target_pose.position)).basicPoint());
+  return arc_coords.distance;
+}
```

Comment:
memo:
This function may not calculate the value correctly before the ego enters the side lane of the pull over lane.
we need to improve that

![image](https://github.com/autowarefoundation/autoware.universe/assets/39142679/bc53f6e1-7a86-479f-b40d-d9f10fcfcabe)

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/6369#discussion_r1485573265

![image](https://github.com/autowarefoundation/autoware.universe/assets/39142679/cae1acf4-2d84-4a00-a26f-5ea37064d5e5)

https://github.com/autowarefoundation/autoware.universe/assets/39142679/e71562d9-5cbc-4041-84d2-a1f4729a04b8